http4k-core/src/main/kotlin/org/http4k/core/Uri.kt

```diff
@@ -30,12 +30,16 @@ data class Uri(val scheme: String, val userInfo: String, val host: String, val p
         }
     }
 
-    val authority = StringBuilder()
-        .appendIfNotBlank(userInfo, userInfo, "@")
+    val hostAndPort = StringBuilder()
         .appendIfNotBlank(host, host)
         .appendIfPresent(port, ":", port.toString())
         .toString()
 
+    val authority = StringBuilder()
+        .appendIfNotBlank(userInfo, userInfo, "@")
+        .append(hostAndPort)
+        .toString()
```

Comment:
I considered making them `lazy`, but I suppose it's not a big difference performance-wise.